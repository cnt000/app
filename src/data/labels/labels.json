{
  "openMenuLabel": "apri il menu",
  "title": "Pungilandia",
  "titleDescription": "Piante grasse, strane ed insolite",
  "goToCartLabel": "vai al carrello",
  "copyright": "Copyright © 2020",
  "privacy": "privacy",
  "contact": "contatti",
  "company": "Azienda Agricola Pungilandia Rocchi Domenico",
  "companyAddress": "Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B\nIscrizione REA Pescara Nr.97422",
  "disclaimer": "Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse\nNegli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9\nDicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001\nVAB",
  "mailToLink": "mailto:info@pungilandia.com",
  "privacyLink": "http://www.pungilandia.com",
  "addToCart": "Aggiungi al carrello",
  "searchPlants": "Cerca piante",
  "searchPlantsImage": "http://www.pungilandia.com/php/images/lo/cotyl_mediop.jpg",
  "hintsFromPungilandiaTitle": "I Consigli di Pungilandia",
  "articleBy": "A cura del dott. Mimmo Rocchi",
  "articleContent": "consigliColtivazione.md",
  "mimmoImage": "http://www.pungilandia.com/media/MIMMO.JPG",
  "menuChiSiamo": "Chi siamo",
  "menuListaPiante": "Lista piante",
  "menuNovita": "Novità",
  "menuProdottiSpeciali": "Prodotti speciali",
  "menuConsigliPerColtivare": "Consigli per coltivare",
  "menuOrdini": "Ordini",
  "menuFotoShow": "Foto show",
  "menuEmail": "E-mail"
}
