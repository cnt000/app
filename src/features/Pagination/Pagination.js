import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import styles from './Pagination.module.css';

const Pagination = ({ page, isSearch, searchQuery, totalPages }) => {
  const hasPages = page < totalPages;
  const prevPageLink = isSearch
    ? `/search/${page}?q=${searchQuery}`
    : `/plp/${page}`;
  const nextPageLink = isSearch
    ? `/search/${page + 2}?q=${searchQuery}`
    : `/plp/${page + 2}`;
  return (
    hasPages && (
      <div className={styles.pagination}>
        {page > 0 && (
          <span>
            <Link to={prevPageLink}>Pagina prec.({page})</Link>
          </span>
        )}
        <span className={styles.page}>Pagina {page + 1}</span>
        {page < totalPages - 1 && (
          <span>
            <Link to={nextPageLink}>Pagina succ.({page + 2})</Link>
          </span>
        )}
      </div>
    )
  );
};

Pagination.propTypes = {
  page: PropTypes.number,
  isSearch: PropTypes.bool,
  searchQuery: PropTypes.string,
};

export default Pagination;
