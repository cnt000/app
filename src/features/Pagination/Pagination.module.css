.pagination {
  text-align: center;
  padding: 0.4rem;
  margin: 0.2rem 0;
  height: 36px;
  line-height: 1.6;
}

.page {
  width: 5rem;
  display: inline-block;
}
