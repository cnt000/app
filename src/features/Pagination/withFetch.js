import React from 'react';
import { useParams } from 'react-router-dom';
import { conf } from '../../conf/url';
import { useQuery } from '../../hook/useQuery';
import useFetch from 'use-http';

const withFetch = (Component) => ({ children, ...props }) => {
  const { id, q } = useParams();
  const isSearch = q !== '';
  const query = useQuery();
  const searchQuery = query.get('q');
  const searchQueryWithQs = `?q=${searchQuery}`;
  const url = `${conf.apiEndpoint}${conf.totalProductsUrl}${
    searchQuery ? searchQueryWithQs : ''
  }`;
  const page = id ? Math.max(id - 1, 0) : 0;
  const { loading, data = 0 } = useFetch(url, { persist: true }, []);
  const totalPages = Math.ceil(data / 24);
  return (
    <>
      {loading && 'Fetching...'}
      {!loading && (
        <Component
          isSearch={isSearch}
          page={page}
          searchQuery={searchQuery}
          totalPages={totalPages}
        >
          {children}
        </Component>
      )}
    </>
  );
};

export default withFetch;
