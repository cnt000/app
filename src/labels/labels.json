{
  "openMenuLabel": "apri il menu",
  "title": "Pungilandia",
  "titleDescription": "Piante grasse, strane ed insolite",
  "goToCartLabel": "vai al carrello",
  "copyright": "Copyright Â© 2020",
  "privacy": "privacy",
  "contact": "contatti",
  "company": "Azienda Agricola Pungilandia Rocchi Domenico",
  "companyAddress": "Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B<br>Iscrizione REA Pescara Nr.97422",
  "disclaimer": "Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse<br>Negli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9<br>Dicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001<br>VAB"
}
