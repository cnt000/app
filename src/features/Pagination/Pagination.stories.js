import React from 'react';

import Pagination from './Pagination';
import { withLabels } from '../Labels/withLabels';
import { withRouter } from '../Router/withRouter';

export default {
  component: Pagination,
  title: 'Pagination',
};
const firstPage = () => (
  <Pagination isSearch={false} page={0} searchQuery={'agave'} />
);

const paginationSearch = () => (
  <Pagination isSearch={true} page={1} searchQuery={'nigra'} />
);

const lastPage = () => (
  <Pagination isSearch={true} page={4} searchQuery={'nigra'} />
);

export const FirstPage = withLabels(withRouter(firstPage));
export const Search = withLabels(withRouter(paginationSearch));
export const LastPage = withLabels(withRouter(lastPage));
