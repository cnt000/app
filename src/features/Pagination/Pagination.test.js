import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Link } from 'react-router-dom';

import Pagination from './Pagination';
import LabelsContext from '../Labels';
import labels from '../../data/labels/labels.json';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <LabelsContext.Provider value={labels}>
        <Router>
          <Pagination isSearch={false} page={0} searchQuery={'agave'} />
        </Router>
      </LabelsContext.Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
