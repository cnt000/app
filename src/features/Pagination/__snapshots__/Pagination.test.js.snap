// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="pagination"
>
  <span
    className="page"
  >
    Pagina 
    1
  </span>
  <span>
    <a
      href="/plp/2"
      onClick={[Function]}
    >
      Pagina succ.(
      2
      )
    </a>
  </span>
</div>
`;
